dataprofiler/data_readers/data_utils.py

```diff
@@ -602,7 +648,7 @@ def load_as_str_from_file(
 
             # Return either the last index of sample_lines OR
             # the index of the newline char that matches remaining_lines
-            search_query_value = "\n"
+            search_query_value: Union[str, bytes] = "\n"
             if isinstance(sample_lines, bytes):
                 search_query_value = b"\n"
```

Comment:
does it make more sense to cast to string immediately? On line 653 could we do `search_query_value = case(str, b"\n")` then we don't need line 660 and search_query_value can just be statically typed as `str`

Reply:
Instead of casting, could nth loc just accept bytes?

Reply:
@Sanketh7

Reply:
We could make `find_nth_loc` accept bytes but then we'd have to deal with the cases where the search string is of type `str` and the search query is of type `bytes` (and vice versa). One way I found to deal with this is to use the type `AnyStr` in `find_nth_loc` which basically creates a generic type that can only be `str` or `bytes`. However, the issue I run into is that the current way the code is set up, we would have to pass in a `Union[str, bytes]` to `find_nth_loc` but you can't pass that in for `AnyStr` (see https://github.com/python/mypy/issues/1533).